Surface_mesh_topology/include/CGAL/Surface_mesh_topology/internal/Minimal_quadrangulation.h

```diff
@@ -476,6 +524,80 @@ class Minimal_quadrangulation
 #endif // defined(CGAL_PWRLE_TURN_V2) || defined(CGAL_PWRLE_TURN_V3)
   }
 
+  /// @return true iff 'p' is a simple curve.
+  bool is_homotopic_to_simple_cycle(const Path_on_surface<Mesh>& p,
+                       bool display_time=false) const
+  {
+    if (p.is_empty())
+    { return true; }
+
+    if (!p.is_closed())
+    {
+      std::cerr<<"Error: is_homotopic_to_simple_cycle requires a closed path."<<std::endl;
```

Comment:
Shouldn't you have a verbose mode instead of always dumping on cerr?

Reply:
@gdamiand Should we change the implementation of `is_contractible()` as well? Since it seems that many methods in the package does some dumping on cerr thing.

Reply:
For me, both solutions are ok; I don't think this is very important. @johnmave126 Do what you prefer and what you think is better. Thanks.

Reply:
If we were to make a verbose mode, should it be an additional runtime argument or a compile-time macro? Currently there is a runtime argument `display_time` that makes methods report time used and a compile-time macro `CGAL_TRACE_CMAP_TOOLS` that sometimes prints information for the underlying map. I am also not sure whether adding a runtime argument might be a breaking change though.

Reply:
The CGAL_TRACE_CMAP_TOOLS is for us (i.e. developers of the package, to trace and debug our code).

If you want the verbose mode runtime without changing the api of the methods, you can add a boolean verbose_mode in the class, one setter to change its value, and test this boolean to display more information.

Reply:
Addressed in https://github.com/CGAL/cgal/pull/4898/commits/b58c05eea0a5d8ae3013185c90596a82096b3c6a